products/compute/api.yaml

```diff
@@ -9324,6 +9324,179 @@ objects:
+    identity:
+      - name
+    nested_query: !ruby/object:Api::Resource::NestedQuery
+      modify_by_patch: true
+      keys:
+        - bgpPeers
+    description: |
+      BGP information that must be configured into the routing stack to
+      establish BGP peering. This information must specify the peer ASN
+      and either the interface name, IP address, or peer IP address.
+      Please refer to RFC4273.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Google Cloud Router': 'https://cloud.google.com/router/docs/'
+      api: 'https://cloud.google.com/compute/docs/reference/rest/v1/routers'
+    async: !ruby/object:Api::Async
+      operation: !ruby/object:Api::Async::Operation
+        kind: 'compute#operation'
+        path: 'name'
+        base_url: 'projects/{{project}}/regions/{{regions}}/operations/{{op_id}}'
+        wait_ms: 1000
+        timeouts: !ruby/object:Api::Timeouts
+          insert_minutes: 10
+          update_minutes: 10
+          delete_minutes: 10
+      result: !ruby/object:Api::Async::Result
+        path: 'targetLink'
+      status: !ruby/object:Api::Async::Status
+        path: 'status'
+        complete: 'DONE'
+        allowed:
+          - 'PENDING'
+          - 'RUNNING'
+          - 'DONE'
+      error: !ruby/object:Api::Async::Error
+        path: 'error/errors'
+        message: 'message'
+    parameters:
+      - !ruby/object:Api::Type::ResourceRef
+        name: 'router'
+        resource: 'Router'
+        imports: 'name'
+        description: |
+          The name of the Cloud Router in which this BgpPeer will be configured.
+        required: true
+        input: true
+        url_param_only: true
+      - !ruby/object:Api::Type::ResourceRef
+        name: region
+        resource: Region
+        imports: name
+        description: Region where the router and BgpPeer reside.
+        input: true
+        required: true
+        url_param_only: true
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          Name of this BGP peer. The name must be 1-63 characters long,
+          and comply with RFC1035. Specifically, the name must be 1-63 characters
+          long and match the regular expression [a-z]([-a-z0-9]*[a-z0-9])? which
+          means the first character must be a lowercase letter, and all
+          following characters must be a dash, lowercase letter, or digit,
+          except the last character, which cannot be a dash.
+        required: true
+        input: true
+      - !ruby/object:Api::Type::String
+        name: 'interfaceName'
+        description: |
+          Name of the interface the BGP peer is associated with.
+        required: true
+        input: true
+      - !ruby/object:Api::Type::String
+        name: 'ipAddress'
+        description: |
+          IP address of the interface inside Google Cloud Platform.
+          Only IPv4 is supported.
+        input: true
```

Comment:
According to the API documentation for the peer, "This information must specify the peer ASN and either the interface name, IP address, or peer IP address.", hence marking it as `input: true`.

But after testing by only specifying the `peerAsn` and one of the `interfaceName`, `ipAddress` or `peerIpAddress` I can confirm that the `peerIpAddress` is always required, while `ipAddress` and `interfaceName` are always optional, so I'll mark `ipAddress` as output and `peerIpAddress` as required, but leave `interfaceName` as required since we use that to uniquely identify the peer.